vectorstores/alloydb/vectorestores.go

```diff
@@ -0,0 +1,246 @@
+package alloydb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	"github.com/pgvector/pgvector-go"
+	"github.com/tmc/langchaingo/embeddings"
+	"github.com/tmc/langchaingo/schema"
+	"github.com/tmc/langchaingo/vectorstores"
+)
+
+type vectorStore struct {
+	engine              PostgresEngine
+	embedder            embeddings.Embedder
+	collectionName      string
+	embeddingTableName  string
+	collectionTableName string
+	collectionUUID      string
+}
+
+var _ vectorstores.VectorStore = &vectorStore{}
+
+// NewVectorStore creates a new Store with options.
+func NewVectorStore(ctx context.Context, opts ...VectoreStoresOption) (vectorStore, error) {
+	vs, err := ApplyVectorStoreOptions(opts...)
+	if err != nil {
+		return vectorStore{}, err
+	}
+	return vs, nil
+}
+
+// AddDocuments adds documents to the Postgres collection,
+// and returns the ids of the added documents.
+func (vs *vectorStore) AddDocuments(ctx context.Context, docs []schema.Document, options ...vectorstores.Option) ([]string, error) {
+	opts := vectorstores.Options{}
+	if opts.ScoreThreshold != 0 || opts.Filters != nil || opts.NameSpace != "" {
+		return nil, errors.New("vector store unsupported options")
+	}
+
+	docs = deduplicate(ctx, opts, docs)
+
+	texts := make([]string, 0, len(docs))
+	for _, doc := range docs {
+		texts = append(texts, doc.PageContent)
+	}
+
+	embedder := vs.embedder
+	if opts.Embedder != nil {
+		embedder = opts.Embedder
+	}
+
+	vectors, err := embedder.EmbedDocuments(ctx, texts)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(vectors) != len(docs) {
+		return nil, errors.New("number of vectors from embedder does not match number of documents")
+	}
+
+	b := &pgx.Batch{}
+	sql := fmt.Sprintf(`INSERT INTO %s (uuid, document, embedding, cmetadata, collection_id)
```

Comment:
One thing with this vs the Pgvector design is that we want to be able to support a flexible schema. Is that planning to be addressed in a later PR?